lib/safexDb.js

```diff
@@ -244,23 +244,48 @@ export const requestPortfolio = async function(sender, receiver){
 }
 
 
-export const createNewSafe = async function(creator, recipient, encryptedKey, recipentEnc, encryptedData,idx ){
+export const randomGuardians = async (creatorDID, inheritorDID) => {
+    const {threadDb, client} = await getCredentials()
+    const threadId = ThreadID.fromBytes(threadDb)
+    const users = await client.find(threadId, 'Users', {})
+    let guardians = []
+    // console.log("Users", users)
+    let loopValue = 0
+    while(loopValue <=2){
+        const index = Math.floor(Math.random() * users.length)
+        let randomGuardian = users[index]
+        if(creatorDID !== randomGuardian.did && inheritorDID !== randomGuardian.did && !guardians.includes(randomGuardian.did)){
+            guardians.push(randomGuardian.did)
+            loopValue = loopValue + 1
+        }else{
+            loopValue = guardians.length
+        }
+    }
+    console.log(guardians)
+   return guardians
+}
+
+
+export const createNewSafe = async function(creator, inheritor, encryptedKey, recipentEnc, encryptedData,idx ){
     try {
 
         //generate aes key for the user
         const {threadDb, client} = await getCredentials()
         const threadId = ThreadID.fromBytes(threadDb)
 
         const secretShares = shamirs.split(JSON.stringify(recipentEnc), {shares: 3, threshold: 2})
-        console.log(secretShares);
 
-        const guardianOne = await checkEmailExists("guardianOne@gmail.com");
-        const guardianTwo = await checkEmailExists("guardianTwo@gmail.com");
-        const guardianThree = await checkEmailExists("guardianThree@gmail.com");
 
-        const shardOne = await idx.ceramic.did.createDagJWE(secretShares[0], [guardianOne.user.did]);
-        const shardTwo = await idx.ceramic.did.createDagJWE(secretShares[1], [guardianTwo.user.did]);
-        const shardThree = await idx.ceramic.did.createDagJWE(secretShares[2], [guardianThree.user.did]);
+        const guardians = await randomGuardians(creator, inheritor)
+
+        const guardianOne = await getLoginUser(guardians[0]);
```

Comment:
Better to iterate and store the guardians on a list. It will reduce the below code while storing the safe as well.